### PR TITLE
Deepcore corrections

### PR DESCRIPTION
**Pull request by AsciiSquid**
## About The Pull Request
Moves deepcore stuff around to be more compatible with things that aren't lavaland.

## Why It's Good For The Game
Fixes bugs and more compatible with things that aren't lavaland.

## Changelog
:cl:
add: Added part component support for deepcore Hubs, Hoppers.
balance: make DCM less overwhemingly busted
code: deepcore machines now use material containers
refactor: refactored ore_vein areas to inhabit the landmark code entirely
/:cl:

### CODE REVIEW

File: _maps/_basemap.dm [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// #define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
+#define LOWMEMORYMODE //uncomment this to load centcom and runtime station and thats it.
```
Comment (Maldaris) [inline _maps/_basemap.dm]: Comment this back out

File: waspstation/code/modules/mining/deepcore/ore_vein.dm [hunk 2]
```diff
@@ -0,0 +1,37 @@
+GLOBAL_LIST_EMPTY(ore_vein_landmarks)
+
+/obj/effect/landmark/ore_vein
```
Comment (Maldaris) [inline waspstation/code/modules/mining/deepcore/ore_vein.dm:3]: In the event we want to create ruins with specific ore veins, subclass this with one of each ore type.

File: waspstation/code/modules/mining/deepcore/ore_vein.dm [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	name = "ore vein"
+	var/datum/material/resource
+	var/material_rate = 0
+
+/obj/effect/landmark/ore_vein/Initialize(mapload, var/datum/material/mat)
+	. = ..()
+	GLOB.ore_vein_landmarks += src
+	// Key = Material path; Value = Material Rate
+	//! Ensure material datum has an ore_type set
+	var/static/list/ores_list = list(
```
Comment (Maldaris) [inline waspstation/code/modules/mining/deepcore/ore_vein.dm:13]: This setup + the way these are spawned looks to me like it doesn't necessarily guarantee that one of each type spawns. Is that accurate?

Comment (AsciiSquid) [reply waspstation/code/modules/mining/deepcore/ore_vein.dm:13]: Correct, however the number of veins spawn near ~100 so we can probably just assume a even distribution.

**Pull request merged**